docs/specification.md

```diff
@@ -1,153 +1,141 @@
 
-## API
+## Routines
 
-Methods are prefixed using the following convention:
+### Signing
 
-- `engine` prefix indicates calls made to the execution client. These methods are specified in the [execution engine APIs](https://github.com/ethereum/execution-apis/blob/v1.0.0-alpha.8/src/engine/specification.md).
-- `builder` prefix indicates calls made to the mev-boost sidecar.
-- `relay` prefix indicates calls made to a relay.
+All signature operations should follow the [standard BLS operations][bls] interface defined in `consensus-specs`.
 
-### engine_forkchoiceUpdatedV1
+There are two types of data to sign over in the Builder API:
+* in-protocol messages, e.g. [`BeaconBlockHeader`][beacon-block-header], which should compute the signing root using [`computer_signing_root`][compute-signing-root] and use the domain specified for beacon block proposals.
+* Builder API messages, e.g. [`builder_setFeeRecipientV1`](#builder_setFeeRecipientV1), which should compute the signing root using [`compute_signing_root`][compute-signing-root] and the domain `DomainType('0xXXXXXXXX')` (TODO: get a proper domain).
 
-See [engine_forkchoiceUpdatedV1](https://github.com/ethereum/execution-apis/blob/v1.0.0-alpha.8/src/engine/specification.md#engine_forkchoiceupdatedv1).
+As `compute_signing_root` takes `SSZObject` as input, client software should convert in-protocol messages to their SSZ representation to compute the signing root and Builder API messages to the SSZ representations defined below. See [`consensus-specs`][consensus-specs] for the definitions of [`Bytes20`][bytes20], [`ExecutionPayloadHeader`][execution-payload-header], and [`uint256`][uint256].
 
-### builder_proposeBlindedBlockV1
-
-#### Request
+#### `feeRecipient`
 
-- method: `builder_proposeBlindedBlockV1`
-- params:
-  1. [`SignedBlindedBeaconBlock`](#signedblindedbeaconblock)
+```python
+feeRecipient = Bytes20
+```
 
-#### Response
+#### `builder_getExecutionPayloadV1` Response
 
-- result: [`ExecutionPayloadV1`](https://github.com/ethereum/consensus-specs/blob/v1.1.10/specs/bellatrix/beacon-chain.md#executionpayload).
-- error: code and message set in case an exception happens while proposing the payload.
+```python
+class Response(Container):
+    payload: ExecutionPayloadHeader
+    value: uint256
+```
 
-Technically, this call only needs to return the `transactions` field of [`ExecutionPayloadV1`](https://github.com/ethereum/consensus-specs/blob/v1.1.10/specs/bellatrix/beacon-chain.md#executionpayload), but we return the full payload for simplicity.
+## Methods
 
-### builder_getPayloadHeaderV1
+### `builder_setFeeRecipientV1`
 
 #### Request
 
-- method: `builder_getPayloadHeaderV1`
+- method: `builder_setFeeRecipientV1`
 - params:
-  1. `payloadId`: `DATA`, 8 Bytes - Identifier of the payload build process
+  1. `feeRecipient`: `DATA`, 20 Bytes - Address of account which should receive fees.
+  2. `signature`: `DATA`, 96 Bytes - Validator signature over `feeRecipient`.
```

Comment:
Is this signature only on the `feeRecipient`? This seems very unsafe since it's easy to leak and masquerade as another validator. Adding either some id for the relay and/or a timing component would fix this I think.

Reply:
There is simply no way to recover a public key in BLS schema. It should be a part of signed data.

Reply:
A timestamp would be good! So the relays can build for the latest sent one.

Reply:
@jparyani how would this allow someone to masquerade as another validator? The builder would map `feeRecipient` to the validator's public key. Since the pk is verified by the signature, there is no malleability. People on other IPs could resubmit the message, but there shouldn't be an issue with that.

@mkalinin thanks for noting this, keep forgetting to add that in.

@metachris good suggestion, I will update to include this.